common/persistence/serialization/task_serializer.go

```diff
@@ -14,16 +14,28 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+var _ TaskSerializer = (*taskSerializer)(nil)
+
 type (
-	TaskSerializer struct {
+	TaskSerializer interface {
+		ReplicationTaskSerializer
+		SerializeTask(task tasks.Task) (*commonpb.DataBlob, error)
+		DeserializeTask(category tasks.Category, blob *commonpb.DataBlob) (tasks.Task, error)
+	}
+	ReplicationTaskSerializer interface {
+		SeralizeReplicationTask(task tasks.Task) (*persistencespb.ReplicationTaskInfo, error)
+		DeserializeReplicationTask(replicationTask *persistencespb.ReplicationTaskInfo) (tasks.Task, error)
+	}
+	taskSerializer struct {
+		serializer Serializer
 	}
 )
 
-func NewTaskSerializer() *TaskSerializer {
-	return &TaskSerializer{}
+func NewTaskSerializer(serializer Serializer) TaskSerializer {
```

Comment:
Can you provide more context on this change?

Reply:
Sure! So before you'll see calls to `TimerTaskInfoFromBlob(blob)` etc in `TaskSerializer`. But these are not methods. So to invoke them now, you need a `Serializer`. Therefore it's `s.serializer.TimerTaskInfoFromBlob(blob)` now.